src/analysis/special_functions/pow.lean

```diff
@@ -1376,6 +1413,20 @@ begin
           nnreal.rpow_le_rpow_of_exponent_ge (bot_lt_iff_ne_bot.mpr h) hx1 hyz] }
 end
 
+lemma rpow_pos_of_pos {p : ℝ} {x : ennreal} (hx_pos : 0 < x) (hp_pos : 0 < p) : 0 < x^p :=
+by { rw ←zero_rpow_of_pos hp_pos, exact rpow_lt_rpow hx_pos hp_pos }
+
+lemma rpow_pos {p : ℝ} {x : ennreal} (hx_pos : 0 < x) (hx_ne_top : x ≠ ⊤) : 0 < x^p :=
+begin
+  cases lt_or_le 0 p with hp_pos hp_nonpos,
+  { exact rpow_pos_of_pos hx_pos hp_pos, },
+  { rw [←neg_neg p, rpow_neg, inv_pos],
+    exact rpow_ne_top_of_nonneg (by simp [hp_nonpos]) hx_ne_top, },
+end
+
+lemma rpow_nonneg_of_pos {p : ℝ} {x : ennreal} (hx_nonneg : 0 ≤ x) (hp_pos : 0 < p) : 0 ≤ x^p :=
```

Comment:
If `x` is an ennreal this is always true right?

Reply:
Indeed it is.